jetty-websocket/websocket-servlet/src/main/java/org/eclipse/jetty/websocket/servlet/WebSocketServlet.java

```diff
@@ -53,167 +51,242 @@
  *     &#064;Override
  *     public void configure(WebSocketServletFactory factory)
  *     {
- *         // set a 10 second idle timeout
- *         factory.getPolicy().setIdleTimeout(10000);
- *         // register my socket
- *         factory.register(MyEchoSocket.class);
+ *       factory.setDefaultMaxFrameSize(4096);
+ *       factory.addMapping(factory.parsePathSpec("/"), (req,res)-&gt;new EchoSocket());
  *     }
  * }
  * </pre>
  * <p>
- * Note: that only request that conforms to a "WebSocket: Upgrade" handshake request will trigger the {@link WebSocketCreatorMapping} handling of creating
- * WebSockets.<br>
- * All other requests are treated as normal servlet requests.
- * <p>
- * <p>
- * <b>Configuration / Init-Parameters:</b><br>
+ * Only request that conforms to a "WebSocket: Upgrade" handshake request will trigger the {@link WebSocketMapping} handling of creating
+ * WebSockets.  All other requests are treated as normal servlet requests.  The configuration defined by this servlet init parameters will
+ * be used as the customizer for any mappings created by {@link WebSocketServletFactory#addMapping(PathSpec, WebSocketCreator)} during
+ * {@link #configure(WebSocketServletFactory)} calls.  The request upgrade may be peformed by this servlet, or is may be performed by a
+ * {@link WebSocketUpgradeFilter} instance that will share the same {@link WebSocketMapping} instance.  If the filter is used, then the
+ * filter configuraton is used as the default configuration prior to this servlets configuration being applied.
+ * </p>
  * <p>
+ * <b>Configuration / Init-Parameters:</b>
+ * </p>
  * <dl>
  * <dt>maxIdleTime</dt>
  * <dd>set the time in ms that a websocket may be idle before closing<br>
- * <p>
  * <dt>maxTextMessageSize</dt>
  * <dd>set the size in UTF-8 bytes that a websocket may be accept as a Text Message before closing<br>
- * <p>
  * <dt>maxBinaryMessageSize</dt>
  * <dd>set the size in bytes that a websocket may be accept as a Binary Message before closing<br>
- * <p>
  * <dt>inputBufferSize</dt>
- * <dd>set the size in bytes of the buffer used to read raw bytes from the network layer<br>
+ * <dd>set the size in bytes of the buffer used to read raw bytes from the network layer<br> * <dt>outputBufferSize</dt>
+ * <dd>set the size in bytes of the buffer used to write bytes to the network layer<br>
+ * <dt>maxFrameSize</dt>
+ * <dd>The maximum frame size sent or received.<br>
+ * <dt>autoFragment</dt>
+ * <dd>If true, frames are automatically fragmented to respect the maximum frame size.<br>
  * </dl>
  */
 @SuppressWarnings("serial")
 public abstract class WebSocketServlet extends HttpServlet
 {
+    // TODO This servlet should be split into an API neutral version and a Jetty API specific one.
```

Comment:
This is already neutral and works with all APIs, it's not Jetty API specific.

Reply:
It is mostly neutral, but the `WebSocketServlet.CustomizedWebSocketServletFactory#addMapping` assumes a FrameHandlerFactory and probably only works with the jetty one....  So it is neutral, but in a fragile way.

Reply:
see also your comment on `register` above for non-neutrality